filebeat/beater/filebeat.go

```diff
@@ -195,14 +196,16 @@ func (fb *Filebeat) setupPipelineLoaderCallback(b *beat.Beat) error {
 
 	overwritePipelines := true
 	b.OverwritePipelinesCallback = func(esConfig *conf.C) error {
-		esClient, err := eslegclient.NewConnectedClient(esConfig, "Filebeat")
+		ctx, cancel := context.WithCancel(context.Background())
```

Comment:
Have you considered to use context with timeout here instead? And maybe to have the value of timeout configurable?

Reply:
That means adding a new feature, I believe it's a good idea, however it feels out of the scope of this PR. The focus of this PR is to refactor the code and add tests to ensure https://github.com/elastic/beats/issues/40705 won't happen again.

So I rather not add it on this PR.

Reply:
In this context (no pun intended) I would rather use context.TODO(), but keeping it as-is is also fine.